cg/apps/lims/api.py

```diff
@@ -388,3 +388,32 @@ def get_sample_comment(self, sample_id: str) -> str:
     def get_sample_project(self, sample_id: str) -> str:
         """Get the lims-id for the project of the sample"""
         return self.sample(sample_id).get("project").get("id")
+
+    def get_sample_rin(self, sample_id: str) -> float:
+        """Get the sample RIN value determined in the Initial QC steps"""
+        sample_artifact_id = f"{sample_id}PA1"
+        sample_artifact = Artifact(self, id=sample_artifact_id)
+        return sample_artifact.udf.get("RIN")
+
+    def get_rna_input_amount(self, sample_id: str) -> float:
+        """Get the input amount used in the latest prep of an RNA sample."""
+        step_names_udfs = MASTER_STEPS_UDFS["rna_prep_step"]
+
+        input_amounts = []
+
+        for process_type in step_names_udfs:
+            artifacts = self.get_artifacts(
+                samplelimsid=sample_id, process_type=process_type, type="Analyte"
+            )
+
+            udf_key = step_names_udfs[process_type]
+            for artifact in artifacts:
+                input_amounts.append(
+                    (
+                        artifact.parent_process.date_run,
+                        artifact.udf.get(udf_key),
+                    )
+                )
```

Comment:
How about using a utility function here:

```
def _add_rna_input_amounts(artifacts=artifacts) -> List[Tuple[datetime, float]]
```

Reply:
I don't really see a way where a utility function would clear up the code that much. There is currently too much info needed from outside the for loop regarding which process type currently is being looked at. Will however do one for the sorting and extraction of the last used input amount.